crates/cargo-contract/src/cmd/build/mod.rs

```diff
@@ -650,7 +655,29 @@ pub(crate) fn execute(args: ExecuteArgs) -> Result<BuildResult> {
             &crate_metadata.contract_artifact_name,
         )?;
 
-        Ok(optimization_result)
+        let cargo_contract_version =
+            if let Ok(version) = Version::parse(env!("CARGO_PKG_VERSION")) {
```

Comment:
AFAIU `CARGO_PKG_VERSION` is only available at compile time (and also during `cargo run`). So would need to be captured in a `const` e.g. `const VERSION: &'static str = env!("CARGO_PKG_VERSION");`